addons/binding/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/handler/DLinkMotionSensorHandler.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dlinksmarthome.handler;
+
+import static org.openhab.binding.dlinksmarthome.DLinkSmartHomeBindingConstants.HARDWARE_VERSION;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.dlinksmarthome.internal.motionsensor.DLinkMotionSensorCommunication;
+import org.openhab.binding.dlinksmarthome.internal.motionsensor.DLinkMotionSensorCommunication.DeviceStatus;
+import org.openhab.binding.dlinksmarthome.internal.motionsensor.DLinkMotionSensorConfig;
+import org.openhab.binding.dlinksmarthome.internal.motionsensor.DLinkMotionSensorListener;
+
+/**
+ * The {@link DLinkMotionSensorHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Mike Major - Initial contribution
+ */
+public class DLinkMotionSensorHandler extends BaseThingHandler implements DLinkMotionSensorListener {
+
+    private static final String MOTION = "motion";
+    private static final String REBOOT_INTERVAL = "rebootInterval";
+
+    private DLinkMotionSensorCommunication motionSensor;
+
+    private final ChannelUID motionChannel;
+
+    public DLinkMotionSensorHandler(final Thing thing) {
+        super(thing);
+        motionChannel = new ChannelUID(getThing().getUID(), MOTION);
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // Does not support commands
+    }
+
+    @Override
+    public void initialize() {
+        if (getConfig().get(REBOOT_INTERVAL) == null) {
+            final Configuration configuration = editConfiguration();
+            if ("A1".equals(getThing().getProperties().get(HARDWARE_VERSION))) {
+                configuration.put(REBOOT_INTERVAL, 4);
+            } else {
+                configuration.put(REBOOT_INTERVAL, 24);
+            }
+            updateConfiguration(configuration);
+        }
+
+        final DLinkMotionSensorConfig config = getConfigAs(DLinkMotionSensorConfig.class);
+        motionSensor = new DLinkMotionSensorCommunication(config, this, scheduler);

```

Comment:
Originally I used my own thread, but spotted this wasn't following the coding guidelines so it got changed to passing the scheduler.

The binding works and has been in use for over a month now so I would prefer not to have to rewrite parts of it.